contracts/automation/andromeda-task-balancer/src/contract.rs

```diff
@@ -0,0 +1,300 @@
+) -> Result<Response, ContractError> {
+    let state = State {
+        contracts: Uint128::zero(),
+        max: msg.max,
+        storage_code_id: msg.storage_code_id,
+        admin: info.sender.to_string(),
+    };
+
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+    STATE.save(deps.storage, &state)?;
+    ADOContract::default().instantiate(
+        deps.storage,
+        env,
+        deps.api,
+        info,
+        BaseInstantiateMsg {
+            ado_type: "task-balancer".to_string(),
+            ado_version: CONTRACT_VERSION.to_string(),
+            operators: None,
+            modules: None,
+            primitive_contract: None,
+        },
+    )
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn reply(_deps: DepsMut, _env: Env, msg: Reply) -> Result<Response, ContractError> {
+    if msg.result.is_err() {
+        return Err(ContractError::Std(StdError::generic_err(
+            msg.result.unwrap_err(),
+        )));
+    }
+
+    Ok(Response::default())
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    let contract = ADOContract::default();
+    match msg {
+        ExecuteMsg::AndrReceive(msg) => contract.execute(deps, env, info, msg, execute),
+        ExecuteMsg::Add { contract } => try_add(deps, env, info, contract),
+        ExecuteMsg::UpdateAdmin { new_admin } => try_update(deps, info, new_admin),
+        ExecuteMsg::RemoveProcess { process_address } => {
+            remove_process(deps, env, info, process_address)
+        }
+    }
+}
+
+fn remove_process(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    process_address: String,
+) -> Result<Response, ContractError> {
+    CONTRACTS.remove(deps.storage, process_address.clone());
+    Ok(Response::new()
+        .add_attribute("action", "removed_process")
+        .add_attribute("process", process_address))
+}
+
+fn try_add(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    contract: String,
+) -> Result<Response, ContractError> {
+    // Not anyone should be allowed to add tasks to tree
+    let state = STATE.load(deps.storage)?;
+    ensure!(info.sender == state.admin, ContractError::Unauthorized {});
+
+    // Task balancing variable creation
+    let mut num = Uint128::from(env.block.height) % state.contracts;
+    let mut count = Uint128::zero();
+
+    loop {
+        // Get contract address from MAP
+        let address = CONTRACTS.load(deps.storage, num.to_string())?;
+
+        // queries specified contract to find if full
+        // Checks if contract is full or not
+        let total: GetSizeResponse = deps.querier.query(&QueryRequest::Wasm(WasmQuery::Smart {
+            contract_addr: address.clone(),
+            msg: to_binary(&LoopQueryMsg::GetSize {})?,
+        }))?;
+        if count > state.contracts {
```

Comment:
This should be done first in the loop to break before any other messages. This will potentially reduce gas fees.